### ISSUE TITLE
How to fill in the IP address of multiple IOCs

### ISSUE THREAD

**Issue by yc593204787** [JeffersonLab/epics2kafka#6]
If I have PVs on multiple IOCs that need to be written to KAFKA, how should the property monitor.addr.list be filled? Spaces between multiple IP addresses? It doesn't seem to support this?

**Comment by slominskir** [comment]
JCA is the EPICS API used so spaces are the delimiter according to the JCA javadocs:  https://www.javadoc.io/doc/org.epics/jca/latest/gov/aps/jca/JCALibrary.html.   If you're using epics2kakfa in a container then be aware that by default containers run in a dedicated isolated network so outside hosts are not reachable unless you go through some network configuration gymnastics.

**Comment by yc593204787** [comment]
Thanks a lot, is there any recommended way to get the timestamp of the PV? Or do I need to use the native build?

**Comment by slominskir** [comment]
I have some notes on timestamps here:  https://github.com/JeffersonLab/epics2kafka/wiki/Developer-Notes#timestamp

All Kafka messages already provide one timestamp in their metadata.  However, there are two types of timestamps in Kafka - application provided (CreateTime) and broker provided (LogAppendTime).  The epics2kafka application uses application provided (in other words, the EPICS IOC provides the timestamp).   In some cases (such as out-of-sync time on IOCs), this may not be good - in that case you can route the messages through a transform and replace with LogAppendTime (or add a new field and have both).

**Comment by yc593204787** [comment]
According to your suggestion, I have obtained the timestamp. Now, I have added the monitor.poll.millis configuration to the ca-source.json file, and set it to 200ms because of the PV refresh frequency, but it doesn't seem to work,the program still obtains the PV value once a second. This attribute cannot be modified?

**Comment by slominskir** [comment]
Are you sure your source IOC is producing messages faster than once a second?

**Comment by yc593204787** [comment]
Yes, for example, a PV refresh rate of DCCT is 5Hz. If the connect is monitored every 1 second, 4 PV data will be lost. So I want to try to modify the value of monitor.poll.millis, but it doesn't seem to work.

**Comment by slominskir** [comment]
I've added a new integration test that demonstrates a fast update.   Run the gradle task "integrationTest" to see it:

```
gradlew integrationTest --tests *BasicIntegrationTest.testFastUpdate
```

 The new [testFastUpdate](https://github.com/JeffersonLab/epics2kafka/blob/24afb7f295606e995f3c253feb1142396fc9bf53/src/integration/java/org/jlab/kafka/connect/integration/BasicIntegrationTest.java#L109) test uses a `"monitor.poll.millis": "200"` [configuration](https://github.com/JeffersonLab/epics2kafka/blob/24afb7f295606e995f3c253feb1142396fc9bf53/examples/integration/connect/ca-source.json#L8).  I've added a [softioc CALC PV](https://github.com/JeffersonLab/epics2kafka/blob/24afb7f295606e995f3c253feb1142396fc9bf53/examples/integration/softioc/softioc.db#L13) named `channelb` that updates once every 200 milliseconds.    The test consumer polls every 200 milliseconds in a loop for 5 seconds and then counts the number of records received.   In a perfectly synchronized system that would mean 25 updates (5 seconds X 5 per second).   However, keep in mind that there are three separate threads looping out-of-sync (IOC thread, connect thread, test consumer thread) so there isn't perfect alignment and the few times I ran the test I got 21 updates in 5 seconds.  Also note if you play around with timing that Kafka buffers messages and many initial polls are empty, especially with longer poll frequency.   I'm not sure why you were not seeing fast updates though - are you using custom consumer code?
